src/main/java/org/craftercms/studio/impl/v2/repository/RepositoryStartupCleanup.java

```diff
@@ -0,0 +1,122 @@
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.craftercms.studio.impl.v2.repository;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import org.craftercms.studio.api.v2.utils.StudioConfiguration;
+import org.craftercms.studio.api.v1.service.security.SecurityService;
+import org.craftercms.studio.api.v2.service.security.internal.UserServiceInternal;
+import org.craftercms.commons.crypto.TextEncryptor;
+import org.craftercms.studio.api.v1.service.GeneralLockService;
+import org.craftercms.studio.api.v2.repository.RetryingRepositoryOperationFacade;
+import org.craftercms.studio.api.v1.service.site.SiteService;
+import org.craftercms.studio.api.v2.utils.GitRepositoryHelper;
+import org.craftercms.studio.impl.v2.utils.GitUtils;
+
+import org.craftercms.studio.api.v1.constant.GitRepositories;
+import static org.craftercms.studio.api.v1.constant.GitRepositories.PUBLISHED;
+import static org.craftercms.studio.api.v1.constant.GitRepositories.SANDBOX;
+
+import org.springframework.beans.factory.annotation.Required;
+
+/**
+ * Clean up git repositories on startup
+ *
+ * @author Phil Nguyen
+ */
+
+public class RepositoryStartupCleanup {
+    private static final Logger logger = LoggerFactory.getLogger(RepositoryStartupCleanup.class);
+
+    protected SiteService siteService;
+    protected StudioConfiguration studioConfiguration;
+    protected SecurityService securityService;
+    protected UserServiceInternal userServiceInternal;
+    private TextEncryptor encryptor;
+    protected GeneralLockService generalLockService;
+    protected RetryingRepositoryOperationFacade retryingRepositoryOperationFacade;
+
+    public void cleanup() {
+        logger.debug("Clean up git lock for all repositories.");
+        try {
+            GitRepositoryHelper helper =  GitRepositoryHelper.getHelper(studioConfiguration, securityService,
+                    userServiceInternal, encryptor, generalLockService, retryingRepositoryOperationFacade);
+            siteService.getAllAvailableSites().forEach(siteId -> {
+                logger.debug("Clean up git lock for site {}", siteId);
+                cleanupSite(siteId, SANDBOX, helper);
+                cleanupSite(siteId, PUBLISHED, helper);
+            });
+        } catch (Exception e) {
+            logger.error("Error cleaning up git lock", e);
+        }
+    }
+
+    protected void cleanupSite(String siteId, GitRepositories repository, GitRepositoryHelper helper) {
+        Path repoPath = helper.buildRepoPath(repository, siteId);
```

Comment:
Should we use the GeneralizedLockingService to lock the repos before we manipulate them?

Reply:
Added lock. Thank you!